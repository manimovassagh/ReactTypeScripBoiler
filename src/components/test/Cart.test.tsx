import React from 'react'
import {render, act, getNodeText, RenderResult, books, matchSnapshot} from './test-utils';
import Cart from '../Cart'
// import {screen} from '@testing-library/dom'
import userEvent from '@testing-library/user-event'

describe('BookDetails rendering and button clicks', () => {
  let documentBody: RenderResult
  beforeEach(async () => {
    await act(async () => {documentBody = render(<Cart />, {store: {cart: [books[0]]}})})
    // screen.debug()
  })

  test('renders', () => {
    expect(documentBody.getByText(/Shopping Cart/i)).toBeInTheDocument();
  })

  test('matches snapshot', () => matchSnapshot(documentBody));

  test('counter goes up', () => {
    const counter = documentBody.getByTestId(/count-target/i)
    expect(getNodeText(counter)).toBe("1")
    userEvent.click(
      documentBody.getByText(/Add One/i)
    )
    expect(getNodeText(counter)).toBe("2")
  })

  test('counter goes down', () => {
    userEvent.click(
      documentBody.getByText(/Remove One/i)
    )
    expect(documentBody.queryAllByTestId(/count-target/i).length).toBe(0)
  })
})
