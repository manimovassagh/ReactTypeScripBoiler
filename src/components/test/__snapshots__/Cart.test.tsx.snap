// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetails rendering and button clicks matches snapshot 1`] = `
<body>
  <div>
    <h2>
      Shopping Cart
    </h2>
    <div
      class="ui middle aligned selection divided list"
    >
      <a
        class="item"
        href="/books/9783864906466"
      >
        <img
          alt=""
          class="ui tiny image"
          src="https://api3.angular-buch.com/images/angular_auflage2.jpg"
        />
        <div
          class="content"
        >
          <div
            class="header"
          >
            Angular
          </div>
          <div
            class="description"
          >
            Grundlagen, fortgeschrittene Themen und Best Practices â€“ mit NativeScript und NgRx
          </div>
          <div
            class="metadata"
          >
            <span>
              Ferdinand Malcher
              , 
            </span>
            <span>
              Johannes Hoppe
              , 
            </span>
            <span>
              Danny Koppenhagen
            </span>
            <br />
            ISBN 
            9783864906466
          </div>
        </div>
        <div
          class="right floated content"
        >
          <div
            class="ui buttons"
          >
            <label
              class="ui button"
              data-testid="count-target"
            >
              <i
                class="shopping cart icon"
              />
              1
            </label>
            <button
              class="ui button green"
            >
              Add One
            </button>
            <button
              class="ui button red"
            >
              Remove One
            </button>
          </div>
        </div>
      </a>
    </div>
  </div>
</body>
`;
